controllers/gce/utils/utils_test.go

```diff
@@ -0,0 +1,452 @@
+		{"will_be_truncated", "560b5b8154db29b3", "foo-e2e-ingress-firewall-xyttxtps-zsy13-auth--560b5b8154db29b3", "l7--foo-e2e-ingress-firewall-xyttxtps-zsy13-auth--560b5b8154db0"},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			namer := NewNamer(fooTest.cn, fooTest.fn)
+			rfn := namer.FrSuffix()
+			if rfn != fooTest.efn {
+				t.Errorf("Returned %v but expected %v", rfn, fooTest.efn)
+			}
+		})
+	}
+}
+
+func TestFrName(t *testing.T) {
+	fooTests := []struct {
+		tName string
+		s     string
+		efn   string
+	}{
+		{"empty_fireName", "", "k8s-fw-"},
+		{"normal_fireName", "fooFN", "k8s-fw-fooFN"},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			namer := NewNamer("fooCN", "fooFN")
+			rfn := namer.FrName(fooTest.s)
+			if rfn != fooTest.efn {
+				t.Errorf("Returned %v but expected %v", rfn, fooTest.efn)
+			}
+		})
+	}
+}
+
+func TestLBName(t *testing.T) {
+	fooTests := []struct {
+		tName string
+		cn    string
+		k     string
+		elbn  string
+	}{
+		{"empty_clusterNameAndKey", "", "", ""},
+		{"empty_clusterName", "", "default/k8s-tps-e2e-ingress-deployment", "default-k8s-tps-e2e-ingress-deployment"},
+		{"end_with_cluster_name", "560b5b8154db29b3", "default/k8s-tps-e2e-ingress-deployment--560b5b8154db29b3", "default-k8s-tps-e2e-ingress-deployment--560b5b8154db29b3"},
+		{"not_end_with_cluster_name", "560b5b8154db29b3", "default/k8s-tps-e2e-ingress-deployment", "default-k8s-tps-e2e-ingress-deployment--560b5b8154db29b3"},
+		{"will_be_turncted", "560b5b8154db29b3", "default/k8s-tps-e2e-ingress-deployment-xytps-auth", "default-k8s-tps-e2e-ingress-deployment-xytps-auth--560b5b8154d0"},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			namer := NewNamer(fooTest.cn, "fooFN")
+			rlbn := namer.LBName(fooTest.k)
+			if rlbn != fooTest.elbn {
+				t.Errorf("Returned %v but expected %v", rlbn, fooTest.elbn)
+			}
+		})
+	}
+}
+
+func TestGetDefaultBackend(t *testing.T) {
+	var nilBS *compute.BackendService
+	bs := &compute.BackendService{CreationTimestamp: "2017-03-08"}
+
+	fooTests := []struct {
+		tName string
+		gm    GCEURLMap
+		ebs   *compute.BackendService
+	}{
+		{"empty_GCEURLMap", map[string]map[string]*compute.BackendService{}, nilBS},
+		{"include_default_but_empty", map[string]map[string]*compute.BackendService{DefaultBackendKey: {}}, nilBS},
+		{"inluce_other_keys_for_default_backend", map[string]map[string]*compute.BackendService{DefaultBackendKey: {DefaultBackendKey + "_non": bs}}, nilBS},
+		{"defalut_backend_has_been_exist", map[string]map[string]*compute.BackendService{DefaultBackendKey: {DefaultBackendKey: bs}}, bs},
+	}
+
+	for _, fooTest := range fooTests {
+		t.Run(fooTest.tName, func(t *testing.T) {
+			rbs := fooTest.gm.GetDefaultBackend()
+			if rbs != fooTest.ebs {
+				t.Errorf("Returned %v but expected %v", rbs, fooTest.ebs)
+			}
+		})
+	}
+}
+
+func TestString(t *testing.T) {
```

Comment:
First, you have a transient error due to map randomness in the `multi_host` test.

Second, I really don't think this needs to be tested. This is only used for logging.